test/FunctionalTests/Infrastructure/GrpcTestContext.cs

```diff
@@ -0,0 +1,117 @@
+﻿#region Copyright notice and license
+
+// Copyright 2019 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+using System;
+using System.Collections.Concurrent;
+using System.Threading;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using NUnit.Framework;
+
+namespace Grpc.AspNetCore.FunctionalTests.Infrastructure
+{
+    public class GrpcTestContext : IDisposable
+    {
+        private readonly ServiceProvider _serviceProvider;
+        private readonly ConcurrentDictionary<string, ILogger> _serverLoggers;
+        private readonly object _lock = new object();
+
+        public ILoggerFactory LoggerFactory { get; }
+
+        private readonly ILogger _logger;
+
+        public VerifyNoErrorsScope Scope { get; }
+
+        public GrpcTestContext()
+        {
+            _serverLoggers = new ConcurrentDictionary<string, ILogger>(StringComparer.Ordinal);
+
+            var services = new ServiceCollection();
+            services.AddLogging(builder =>
+            {
+                builder.SetMinimumLevel(LogLevel.Trace);
+                builder.AddFilter((s, l) =>
+                {
+                    // Server categories are prefixed with SERVER
+                    var category = s.StartsWith("SERVER ") ? s.AsSpan(7) : s.AsSpan();
+                    if (category.SequenceEqual("Microsoft.AspNetCore.Routing.Matching.DfaMatcher"))
```

Comment:
Why are we special casing this? Is it because it's too noisy?

Reply:
Yeah. I'll add a comment